py/legacypipe/image.py

```diff
@@ -1083,34 +1007,33 @@ def run_psfex(self, git_version=None, ps=None):
             git_version = get_git_version()
         # We write the PSF model to a .fits.tmp file, then rename to .fits
         psfdir = os.path.dirname(self.psffn)
-        psfoutfn = os.path.join(psfdir, os.path.basename(self.sefn).replace('.fits','') + '.fits')
-        psftmpfn = psfoutfn + '.tmp'
-        cmd = 'psfex -c %s -PSF_DIR %s -PSF_SUFFIX .fits.tmp -VERBOSE_TYPE QUIET %s' % (os.path.join(sedir, self.camera + '.psfex'), psfdir, self.sefn)
+        # psfex decides for itself what it's going to name the output file....
+        psftmpfn = os.path.join(psfdir, os.path.basename(self.sefn).replace('.fits','') + '.psf.tmp')
+        cmd = 'psfex -c %s -PSF_DIR %s -PSF_SUFFIX .psf.tmp -VERBOSE_TYPE QUIET %s' % (os.path.join(sedir, self.camera + '.psfex'), psfdir, self.sefn)
         debug(cmd)
         rtn = os.system(cmd)
         if rtn:
             raise RuntimeError('Command failed: %s: return value: %i' % (cmd,rtn))
-        
-        # Update the header
-        hlist = [
-            {'name': 'LEGPIPEV', 'value': git_version, 'comment': "legacypipe git version"},
-            {'name': 'EXPNUM',   'value': self.expnum, 'comment': "exponsure number"},
-            {'name': 'PLVER',    'value': plver,       'comment': "CP version"},
-            {'name': 'PLPROCID', 'value': plprocid,    'comment': "CP processing date hash"},
-            {'name': 'PROCDATE', 'value': procdate,    'comment': "DATE of image file"},
-            {'name': 'IMGDSUM',  'value': datasum,     'comment': "DATASUM of image file"}
-            ]
-        F = fitsio.FITS(psftmpfn, 'rw')
-        F[0].write_keys(hlist)
-        F.close()
-
-        cmd = 'mv %s %s' % (psftmpfn, psfoutfn)
-        rtn = os.system(cmd)
-        if rtn:
-            raise RuntimeError('Command failed: %s: return value: %i' % (cmd,rtn))
-        
-    def run_sky(self, splinesky=False, git_version=None, ps=None, survey=None,
-                gaia=True, release=0):
+
+        # Convert into a "merged psfex" format file.
+        T = psfex_single_to_merged(psftmpfn, self.expnum, self.ccdname)
+        # add our own metadata values
+        for k,v in [
+                ('legpipev', git_version),
+                ('plver',    plver),
+                ('plprocid', plprocid),
+                ('procdate', procdate),
+                ('imgdsum',  datasum),
+                ]:
+            T.set(k, np.array([v]))
+
+        psftmpfn2 = os.path.join(psfdir, os.path.basename(self.sefn).replace('.fits','') + '.psf.tmp2')
+        T.writeto(psftmpfn2)
+        os.remove(psftmpfn)
+        os.rename(psftmpfn2, self.psffn)
+
+    def run_sky(self, splinesky=True, git_version=None, ps=None, survey=None,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [LegacySurveyImage.run_sky is too complex (23)](https://app.codacy.com/app/dstndstn/legacypipe/pullRequest?prid=4730106)